crates/driver/src/domain/liquidity/zeroex.rs

```diff
@@ -1,5 +1,73 @@
-/// A signed 0x Protocol Limit Order [^1].
-///
-/// [^1]: <https://docs.0x.org/limit-orders-advanced-traders/docs/introduction>
+use {
+    crate::domain::eth,
+    anyhow::anyhow,
+    contracts::IZeroEx,
+    ethcontract::Bytes,
+    primitive_types::{H160, H256, U256},
+};
+
 #[derive(Clone, Debug)]
-pub struct LimitOrder {}
+pub struct LimitOrder {
+    pub sell_token: H160,
+    pub buy_token: H160,
+    pub sell_amount: U256,
+    pub buy_amount: U256,
+    pub order: Order,
```

Comment:
`order` already contains sell/buy token/amount, but the values might differ: [1](https://github.com/cowprotocol/services/blob/ebe471630c7df72a9c8ef3acbd7452098b989b1c/crates/solver/src/liquidity/order_converter.rs#L80-L94), [2](https://github.com/cowprotocol/services/blob/f3e38c11629176267358687c22abae99ed94ef7b/crates/solver/src/liquidity/zeroex.rs#L60-L77).

Reply:
How so? I think only [2] is relevant for us (the other one is a CoW native limit order which is represented differently). I think all we need is `Order` and `IZeroEx`. 

Even `full_execution_amount` we should be able to pass along in `to_interaction` (using input/output fields on `solution::interaction::Liquidity`) since this would allow solvers to use this liquidity partially (if we set the full execution amount in to_domain we commit to the amount that needs to be executed before solving).

If I'm not mistaken ([docs](https://docs.0xprotocol.org/en/latest/basics/functions.html#fillorkilllimitorder)), `takerTokenFillAmount` should be what we call `inputs`: 
https://github.com/cowprotocol/services/blob/1647ad8f75a6790e2e7958522769fc12568e743c/crates/driver/src/domain/competition/solution/interaction.rs#L24-L26

Reply:
> If I'm not mistaken ([docs](https://docs.0xprotocol.org/en/latest/basics/functions.html#fillorkilllimitorder)), takerTokenFillAmount should be what we call inputs:

And here, should we also use `input` from Liquidity or order's value?
https://github.com/cowprotocol/services/blob/5184697c8e53d1b1444fd81a3ac14ac7d2e4e6af/crates/driver/src/infra/solver/dto/auction.rs#L212